src/pages/DataSubmissionForm.js

```diff
@@ -41,50 +41,88 @@ export const DataSubmissionForm = () => {
     init();
   }, []);
 
-  let formData = {};
+  const formData = {};
+
+  // separate json object and files to send to consent
+  const separateRegistrationAndFiles = (formData) => {
+    const registration = cloneDeep(formData);
+    const files = {};
+
+    if (!isEmpty(registration.alternativeDataSharingPlanFile)) {
+      files.alternativeDataSharingPlan = registration.alternativeDataSharingPlanFile;
+      delete registration.alternativeDataSharingPlanFile;
+    }
+
+    const consentGroups = registration['consentGroups'];
+
+    for (let i = 0; i < consentGroups?.length; i++) {
+      if (!isEmpty(consentGroups[i].nihInstitutionalCertificationFile)) {
+        files['consentGroups['+i+'].nihInstitutionalCertification'] = registration.nihInstitutionalCertificationFile;
+        delete consentGroups[i].nihInstitutionalCertificationFile;
+      }
```

Comment:
Trying to grok this and not quite getting why we need to do it. I can see the desired need to refrain from uploading duplicate files, but not quite sure we're doing that right. The main form doesn't have an `nihInstitutionalCertificationFile` form field - the only place we have this form element is on the consent group panel. Additionally, what is the value of `nihInstitutionalCertificationFile`? Is it a file name? If so, can it be true that we have two files with the same name (but from different directories) on the user's drive?

Reply:
Thinking more on this. I believe that restricting the user from uploading the same file for multiple consent groups is technically incorrect. A user should be able to use the same file multiple times. We should revisit this on the API side as well as I think we're doing something similar there.

Reply:
These are `File` objects, so they contain information about the filename, etc. All this function does is copy it from the `consentGroups` objects where they are populated by the subcomponents of the form, into their own object so they can be uploaded in a separate form data body part. There is no issue with duplicate files, you can upload the same file, multiple files with the same name, etc. as many times as you want, and that's all good.

Reply:
I updated the code paths so that files have their own `formData` type object, so hopefully it's all a bit clearer. I also uploaded the same file in 3 different places and it all uploaded without a hitch.